packages/portfolio-contract/src/resolver.exo.ts

```diff
@@ -0,0 +1,269 @@
+ */
+
+import { makeTracer, mustMatch } from '@agoric/internal';
+import type { CaipChainId } from '@agoric/orchestration';
+import type { Zone } from '@agoric/zone';
+import { type VowTools, type VowKit, type Vow, VowShape } from '@agoric/vow';
+import { M } from '@endo/patterns';
+import type { Invitation } from '@agoric/zoe';
+
+const trace = makeTracer('Resolver');
+
+export type CCTPTransactionEntry = {
+  chainId: CaipChainId;
+  remoteAddress: `0x${string}`;
+  amountValue: bigint;
+  vowKit: VowKit<void>;
+};
+
+export type ResolverKit = {
+  resolver: {
+    /**
+     * Register a pending CCTP transaction and return a vow that resolves when confirmed.
+     *
+     * @param chainId - CAIP chain ID (e.g., 'eip155:42161')
+     * @param remoteAddress - The remote address receiving the funds
+     * @param amount - The amount being transferred
+     * @returns A vow that resolves when the transaction is confirmed
+     */
+    registerCCTPTransaction: (
+      chainId: CaipChainId,
+      remoteAddress: `0x${string}`,
+      amount: bigint,
+    ) => Vow<void>;
+
+    /**
+     * Confirm a CCTP transaction by providing transaction details.
+     *
+     * @param chainId - CAIP chain ID
+     * @param remoteAddress - The remote address
+     * @param amountValue - The amount value as bigint
+     * @param status - Transaction status ('confirmed' | 'failed' | 'pending')
+     * @returns Success status and message
+     */
+    confirmCCTPTransaction: (
+      chainId: CaipChainId,
+      remoteAddress: `0x${string}`,
+      amountValue: bigint,
+      status: 'confirmed' | 'failed' | 'pending',
+    ) => {
+      success: boolean;
+      message: string;
+      key: string;
+    };
+
+    /**
+     * Get all pending transaction keys for debugging/monitoring.
+     */
+    getPendingTransactions: () => string[];
+  };
+};
+
+export type CCTPConfirmationResult = {
+  success: boolean;
+  message: string;
+  key: string;
+  txDetails: {
+    amount: bigint;
+    remoteAddress: `0x${string}`;
+    status: 'confirmed' | 'failed' | 'pending';
+  };
+  remoteAxelarChain: CaipChainId;
+};
+
+export type ResolverInvitationMakers = {
+  /**
+   * Make an invitation to confirm a CCTP transaction.
+   *
+   * This invitation allows anyone with the proper transaction details
+   * to confirm the completion of a CCTP transaction.
+   */
+  makeConfirmCCTPTransactionInvitation: () => Promise<Invitation>;
+};
+
+/**
+ * Prepare the CCTP Transaction Resolver.
+ *
+ * @param zone - Durable storage zone
+ * @param vowTools - Vow tools for creating promises
+ * @returns A function to create resolver instances
+ */
+export const prepareCCTPResolver = (zone: Zone, vowTools: VowTools) => {
+  const resolverZone = zone.subZone('CCTPResolver');
+  const cctpTransactionRegistry = resolverZone.mapStore<
+    string,
+    CCTPTransactionEntry
+  >('cctpTransactionRegistry');
```

Comment:
the argument to `mapStore` is not a name for the store but a column name for the key (cf. [makeScalarMapStore](https://agoric-sdk.pages.dev/funcs/_agoric_store.index.makeScalarMapStore)).


```suggestion
  >('txKey');
```

as in: "txKey 123 not found" and not "cctpTransactionRegistry 123 not found"

Reply:
oops... maybe it's not... I guess `zone.mapStore(name)` is different from `makeScalarMapStore(tag)`.